### PR TITLE
Enforce lower bound on acceptable node versions

### PR DESCRIPTION
**Pull request by larskuhtz**

### CODE REVIEW

File: src/Chainweb/NodeVersion.hs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+isAcceptedVersion :: NodeVersion -> Bool
+isAcceptedVersion = (<=) minAcceptedVersion
+{-# INLINE isAcceptedVersion #-}
+
+-- -------------------------------------------------------------------------- --
+-- Query Remote Version
+
+requestTimeout :: Int
+requestTimeout = 4 * 1000000
```
Comment (fosskers) [inline src/Chainweb/NodeVersion.hs]: Units?

File: src/Chainweb/NodeVersion.hs [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+-- Chainweb Node Version
+
+newtype NodeVersion = NodeVersion { _getNodeVersion :: [Int] }
+    deriving (Show, Eq, Ord, Generic, NFData)
+
+instance HasTextRepresentation NodeVersion where
+    toText (NodeVersion l) = T.intercalate "." $ sshow <$> l
+    fromText = fmap NodeVersion . traverse treadM . T.splitOn "."
+    {-# INLINE toText #-}
+    {-# INLINE fromText #-}
```
Comment (fosskers) [inline src/Chainweb/NodeVersion.hs]: Consider instead [my versions library](http://hackage.haskell.org/package/versions) for parsing these.

Comment (larskuhtz) [reply src/Chainweb/NodeVersion.hs]: I think it's not needed in this case. We don't have to represent arbitrary version numbers but only chainweb-node version numbers that will always be a lexicographical list of `Int`. Taking a new dependency and using a more complex type seems overkill for that.

Comment (larskuhtz) [reply src/Chainweb/NodeVersion.hs]: I think, the versions library is great for representing heterogenous sets of version from different packages. But here we deal with just a single package and we have full control over the format.

**Pull request merged**